test/sanitize_json.d

```diff
@@ -0,0 +1,231 @@
+import std.typecons : Flag, Yes, No;
+import std.array : Appender, appender;
+import std.bigint;
+import std.internal.cstring : tempCString;
+import std.string : startsWith, replace;
+import std.format : formattedWrite, format;
+import std.file : readText;
+import std.stdio;
+
+import json;
+
+__gshared File outFile;
+
+void usage()
+{
+    writeln("Usage: santize_json <input-json> [<output-json>]");
+}
+int main(string[] args)
+{
+    args = args[1 .. $];
+    if (args.length == 0)
+    {
+        usage();
+        return 1;
+    }
+    string inFilename = args[0];
+    if(args.length == 1)
+    {
+        outFile = stdout;
+    }
+    else if(args.length == 2)
+    {
+        outFile = File(args[1], "w");
+    }
+    else
+    {
+        writeln("Error: too many command line arguments");
+        return 1;
+    }
+
+    string jsonText;
+    {
+        auto inFile = File(inFilename, "r");
+        auto inFileSizeLong = inFile.size;
+        assert(inFileSizeLong < size_t.max, "file too large");
+        size_t inFileSize = cast(size_t)inFileSizeLong;
+        char[] buffer = new char[inFileSize + 1];
+        assert(inFileSize == inFile.rawRead(buffer[0..inFileSize]).length);
+        buffer[inFileSize] = '\0';
+        jsonText = cast(string)buffer;
+    }
+    auto json = parseJson(jsonText, inFilename);
+
+    sanitize(json.array);
+
+    auto writer = Writer(0);
+    writer.writeJson(json);
+    return 0;
+}
+
+void sanitize(JsonValue[] rootArray)
+{
+    foreach (objNode; rootArray)
+    {
+        auto obj = objNode.object;
+        auto kind = obj.getAs!string("kind");
+        if (kind == "compilerInfo")
+            sanitizeCompilerInfo(obj);
+        else if (kind == "buildInfo")
+            sanitizeBuildInfo(obj);
+        else if(kind == "module")
+            sanitizeSyntaxNode(objNode);
+        else if(kind == "semantics")
+            sanitizeSemantics(obj);
+    }
+}
+
+void removeString(JsonValue* value)
```

Comment:
`ref`

Reply:
This is just personal preference, but when a function is modifying a value that's passed into it, I prefer to see some indication of this when calling the function, i.e.
```
// I prefer
removeString(&myValue);
// rather than
removestring(myValue);
```
To me this make it obvious that the function is modifying the value.  Note that I only use this convention for value types.  For example if you had a function that modified an integer as a parameter such as `square`, I would rather have `void square(int* i)` rather than `void square(ref int i)` because I would prefer to see `square(&myInt)` rather than `square(myInt)`.

Reply:
How about `out` then?

See also: https://github.com/dlang/dmd/pull/7754/files

Reply:
That has the same problem doesn't it?  You can't see from the call that the value is being modified.
```D
void squareUsingOut(out int i);
// vs
void squareUsingPtr(int* i);

int i = 100;
squareUsingOut(i); // not obvious that i is being modified
squareUsingPtr(&i); // obvious that square could be modifying i
```